plugin/src/main/java/org/wso2/carbon/uuf/maven/AppMojo.java

```diff
@@ -225,52 +219,48 @@ private DependencyNode getDependencyTree(Set<Artifact> includes) throws MojoExec
         }
     }
 
-    private void createConfigFile(DependencyNode rootNode, String componentsDirectory) throws MojoExecutionException {
-        Configuration configuration = new Configuration();
+    private void createConfigurationFile(DependencyNode rootNode, String componentsDirectory)
+            throws MojoExecutionException {
+        Configuration configuration = new Configuration(parseAppConfigFile());
         // Create the final configuration by traversing through the dependency tree.
         try {
             rootNode.traverse(node -> {
-                String configFilePath = getFilePathIn(node.getArtifactId(), componentsDirectory, FILE_CONFIG);
-                Map configMap;
+                String configFilePath = getFilePathIn(node.getArtifactId(), componentsDirectory, FILE_COMPONENT_CONFIG);
                 // Since we are in a lambda, we throw RuntimeExceptions.
+                ComponentConfig componentConfig;
                 try {
-                    configMap = ConfigurationParser.parse(configFilePath);
+                    componentConfig = ComponentConfigParser.parse(configFilePath);
                 } catch (ParsingException e) {
-                    throw new RuntimeException("Cannot parse '" + FILE_CONFIG + "' of " + node +
+                    throw new RuntimeException("Cannot parse '" + FILE_COMPONENT_CONFIG + "' of " + node +
                                                        " which read from '" + configFilePath + "' path.", e);
                 }
-                if (configMap == null) {
-                    return; // No configuration found for this node.
-                }
                 try {
-                    configuration.merge(configMap);
+                    configuration.merge(componentConfig.getConfig());
                 } catch (IllegalArgumentException e) {
                     throw new RuntimeException(
-                            "Cannot merge configuration Map parsed from '" + FILE_CONFIG + "' of " + node +
+                            "Cannot merge configuration Map parsed from '" + FILE_COMPONENT_CONFIG + "' of " + node +
```

Comment:
I think it looks better if we move the message to the previous line (Because the message will anyway take 2 lines)